frontend/src/components/VaultDetails.vue

```diff
@@ -237,10 +258,20 @@ async function fetchData() {
 }
 
 async function fetchOwnerData() {
-  const vaultKeyJwe = await backend.vaults.accessToken(props.vaultId, true);
-  vaultKeys.value = await loadVaultKeys(vaultKeyJwe);
-  (await backend.vaults.getMembers(props.vaultId)).forEach(member => members.value.set(member.id, member));
-  usersRequiringAccessGrant.value = await backend.vaults.getUsersRequiringAccessGrant(props.vaultId);
+    try {
+      const vaultKeyJwe = await backend.vaults.accessToken(props.vaultId, true);
+      vaultKeys.value = await loadVaultKeys(vaultKeyJwe);
+      (await backend.vaults.getMembers(props.vaultId)).forEach(member => members.value.set(member.id, member));
+      usersRequiringAccessGrant.value = await backend.vaults.getUsersRequiringAccessGrant(props.vaultId);
+    } catch(error) {
+      if (error instanceof ForbiddenError) {
+        vaultRecoveryRequired.value = true;
+      } else {
```

Comment:
> Which call is throwing the ForbiddenException?

It is the `backend.vaults.accessToken(props.vaultId, true)` call.

> If, theoretically, the requests need a long response time and in the meantime i'm no longer an owner, i still get the possibility to recover my access, right?

No, the backend checks in the `/vaults/${vaultId}/access-tokens/${userId}` resource, if the user has the owner role.

> Additionally, can a non-member access this screen somehow (e.g. entering the correct URL)? Because this user would also have the possibility to recover access? Of course, if the user has the recovery key, they can do whatever they want, but for documentation. 🙃

Sure if a user sets `vaultRecoveryRequired.value = true` in the browser debug session, they see the recover vault dialog. If this user also have the recovery key, this user can enter it into the dialog but will fail when executing the recovery because the backend checks the `OWNER` role in the `/{vaultId}/access-tokens/{userId}` resource.

Also, as you have already written, this would only give the user an access token (again), which would result in the ability to unlock the vault. With the recovery key, the user can already unlock the vault.